core/state_processor.go

```diff
@@ -284,16 +284,16 @@ func ApplyUpgrades(c *params.ChainConfig, parentTimestamp *uint64, blockContext
 }
 
 type blockContext struct {
-	number    *big.Int
-	timestamp uint64
+	number *big.Int
+	time   uint64
```

Comment:
note: changing this to time since it should usually be the block.Time() set here. We can revert this field, but the argument to `NewBlockContext` should be `time` imo